crates/swc/src/config/mod.rs

```diff
@@ -425,6 +429,15 @@ impl Options {
             NodeModulesResolver::new(TargetEnv::Node, Default::default(), true),
         );
 
+        let transform_filename = match base {
+            FileName::Real(path) => path.as_os_str().to_str().map(String::from),
+            _ => None,
+        };
+
+        let plugin_context = PluginContext {
+            filename: transform_filename,
```

Comment:
simplified property name to `filename` as I expect there won't be another `file` we'll get confused. 

I choose to pickup only real filename to pass forward. Unsure if we want to forward some other values here instead.